### ISSUE TITLE
Break player down into components

### ISSUE THREAD

**Issue by ac-arcana** [Phazorknight/Cogito#49]
The player class is pretty  large, and includes a lot of concerns. Perhaps player movement, player look, and damage receiver could all be separate components.

**Comment by Phazorknight** [comment]
Yup, definitely a goal of mine to chip away into improving how this is currently set up.

**Comment by aronand** [comment]
I've had a look at the code and at least two things that could be done come to mind:
1. Separate attribute management to its own node
2. Implement a state machine to handle movement

**Comment by Vroodot** [comment]
Current Update for Those Interested:
I recently joined the contribution team and am tackling, at least in part, the State Defined Player Controller (`SDPC`).

For those interested, here's about where the project is at:

Scope: Player Focused Movement State Machine


1. Due to some existing dependencies outside of the Player Controller, there may be some instances where decoupling isn't entirely possible without refactoring other systems.

   In those cases, instead of potentially breaking other systems, I'm moving as much work as possible to 
   the individual states, while the player will likely still be holding onto a handful of bools such as 
   `is_standing`, `is_sprinting`, etc and any associated signals.

2. The State Machine will be primarily Node Based. The Player Controller (`SDPC`) will parent the `SDPCStateMachine` while the SDPCSM will parent the individual `SDPCState`s.


3. Each State is being given its own class. For example, `SDPCStateMove`, `SDPCStateJump`, `SDPCStateSprint` will each extend from `SDPCState`

**Comment by Vroodot** [comment]
Some other contributors have pointed me towards using [Godot State Charts](https://derkork.github.io/godot-statecharts/)

Now that I've a bit of free time, I'll be tackling this section if it is still open.

Ignore the previous comment on how the structure will be and stay tuned for updates 👍

**Comment by Phazorknight** [comment]
Thanks @Vroodot 

I'm actually going to close this issue. It's very old, and now that we have discussions for new features, I'd like to keep issues to specific issues and bugs.
